beacon/weakcoin/weak_coin_test.go

```diff
@@ -358,19 +391,25 @@ func TestWeakCoinEncodingRegression(t *testing.T) {
 		signing.WithKeyFromRand(rng),
 	)
 	require.NoError(t, err)
-	vrfSig, err := signer.VRFSigner(signing.WithNonceForNode(1, signer.NodeID()))
-	require.NoError(t, err)
-
-	vrfVerify, err := signing.NewVRFVerifier(signing.WithNonceForNode(1, signer.NodeID()))
+	vrfSig, err := signer.VRFSigner()
 	require.NoError(t, err)
 
 	allowances := weakcoin.UnitAllowances{string(signer.PublicKey().Bytes()): 1}
-	instance := weakcoin.New(broadcaster, vrfSig, vrfVerify, weakcoin.WithThreshold([]byte{0xff}))
+	instance := weakcoin.New(
+		broadcaster,
+		datastore.NewCachedDB(sql.InMemory(), logtest.New(t)),
+		types.RandomNodeID(),
+		vrfSig,
+		vrfVerifierAdapter{},
+		nonceFetcher(t, ctrl),
+		weakcoin.WithThreshold([]byte{0xff}),
+	)
 	instance.StartEpoch(context.Background(), epoch, allowances)
 	require.NoError(t, instance.StartRound(context.Background(), round))
 
+	// TODO(mafa): figure out why sig is now different on every test run
```

Comment:
but it should change. nonce is an extra data to the vrf message now.

Reply:
The problem was that every time I ran the test the signature would be different (without changing code). I've fixed the issue so this comment is obsolete.

Removed.